### PR TITLE
[docs] - Add Cloud to Deployment, sidenav, landing page

### PR DESCRIPTION
**Pull request by erinkcochran87**
This PR:

- Splits the **Deployment** section into two - one for Cloud, one for 'DIY' - and adds it to the sidenav
- Adds category pages for OS deployment (overall), concepts, and guides
- Adds Cloud (back) to the landing page
- Removes the old Cloud docs link from the sidenav
- Completes two Cloud pages that were blank or missing (AWS ECS agent, Deployment types)

Feel free to leave comments on the Vercel preview

### CODE REVIEW

File: docs/content/deployment/concepts.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# Dagster Deployment Concepts
+
+<Note>
+  This is applicable to Dagster Open Source. If you're looking for Dagster
+  Cloud, <a href="/dagster-cloud">click here</a>.
+</Note>
+
+Learn about the concepts relevant to deploying Dagster.
+
+**Not sure where to start?** Check out the [Dagster architecutre guide](/deployment/overview).
```
Comment (clairelin135) [inline docs/content/deployment/concepts.mdx:14]:
```suggestion
**Not sure where to start?** Check out the [Dagster architecture guide](/deployment/overview).
```

File: docs/content/deployment/open-source.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+Get started with a look at Dagster's architecture, including a primer on deployment concepts. [Learn more](/deployment/overview).
+
+---
+
+## Concepts
+
+Learn about the concepts relevant to deploying Dagster.
+
+<ArticleList>
```
Comment (clairelin135) [inline docs/content/deployment/open-source.mdx:26]: Maybe we can add a link to the concepts page here, with a message like: "for an overview of Dagster's concepts, visit this page"?

File: docs/content/deployment/open-source.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  <ArticleListItem
+    title="Dagster instance"
+    href="/deployment/dagster-instnace"
```
Comment (clairelin135) [inline docs/content/deployment/open-source.mdx:29]:
```suggestion
    href="/deployment/dagster-instance"
```

**Review by clairelin135 — COMMENTED** [review]
I love the new condensed navbar. The only thing is that it feels a little weird that I can't find, for example, the `deployment/dagster-instance` page through the navbar. Would it be overkill to put another level of nesting in the navbar?

**Comment by clairelin135** [comment]
Also, the new documentation on cloud options is super clear and easy to read! Love it!

**Comment by erinkcochran87** [comment]
@clairelin135 Re:

> 
I love the new condensed navbar. The only thing is that it feels a little weird that I can't find, for example, the deployment/dagster-instance page through the navbar. Would it be overkill to put another level of nesting in the navbar?

I didn't do this originally because I thought our setup only allowed for three levels... I swore the tests threw a fit when I tinkered with this before. I just pushed up a single test entry though, so we'll see how it goes!

**Review by clairelin135 — APPROVED** [review]
Nice, I like it! If it's possible, I think it would also be nice to add the fourth level of dropdowns for some of the other sections:
- developing and testing in dagster cloud
- dagster cloud authentication
- understanding deployment types
Doesn't have to happen within this PR though

**Pull request merged**